webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java

```diff
@@ -67,27 +92,156 @@
+                    Site site = siteService.getSite(bean.getSiteId());
+                    bean.setSiteTitle(site.getTitle());
+                    if (bean.getReference() != null && !bean.getReference().startsWith("/user/")) {
+                        entityManager.getUrl(bean.getReference(), Entity.UrlType.PORTAL).ifPresent(u -> bean.setUrl(u));	
+                    }
+                    String taskAssignedTo = getTaskAssignedDescription(bean.getTaskId(), site);
+                    bean.setTaskAssignedTo(taskAssignedTo);
                 } catch (IdUnusedException e) {
                     log.warn("No site for id {}", bean.getSiteId());
                 }
                 return bean;
             }).collect(Collectors.toList());
     }
+    
+    @GetMapping(value = "/tasks/site/{siteId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<UserTaskAdapterBean> getSiteTasks(@PathVariable String siteId) throws UserNotDefinedException {
+
+        checkSakaiSession();
+
+        // Flatten the UserTask objects into a more compact form and return.
+        return taskService.getAllTasksForCurrentUserOnSite(siteId)
+            .stream().map(bean -> {
+                try {
+                    Site site = siteService.getSite(bean.getSiteId());
+                    bean.setSiteTitle(site.getTitle());
+                    if (bean.getReference() != null && !bean.getReference().startsWith("/user/")) {
+                        entityManager.getUrl(bean.getReference(), Entity.UrlType.PORTAL).ifPresent(u -> bean.setUrl(u));
+                    }
+                    String taskAssignedTo = getTaskAssignedDescription(bean.getTaskId(), site);
+                    bean.setTaskAssignedTo(taskAssignedTo);
+                } catch (IdUnusedException e) {
+                    log.warn("No site for id {}", bean.getSiteId());
+                }
+                return bean;
+            }).collect(Collectors.toList());
+    }
+    
+    @GetMapping(value = "/tasks/role/{siteId}/{userId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, String> getCurrentUserRoleOnSite(@PathVariable String siteId, @PathVariable String userId) {
+        checkSakaiSession();
+
+        String roleId = "access";
+        try {
+            roleId = siteService.getSite(siteId).getMember(userId).getRole().getId();
+        } catch (Exception e) {
+            log.warn("Error retrieving role on site {} for user {}", siteId, userId);
+        }
+        return Collections.singletonMap("role", roleId);
+    }
+
+    @GetMapping(value = "/tasks/site/groups/{siteId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Object[] getSiteGroups(@PathVariable String siteId) {
+        checkSakaiSession();
+
+        Map<String, String> groupsMap = new HashMap<>();
+        try {
+            Site currentSite = siteService.getSite(siteId);
+            Collection groups = currentSite.getGroups();
```

Comment:
Why throw your typing away here?

Reply:
Grab the site at the top of the method. If you get an IdUnusedException, throw an IllegalArgumentException (siteId must be invalid). Then you can use Collectors.toMap to build your groups map.